packages/foam-vscode/static/dataviz/graph.js

```diff
@@ -241,10 +274,46 @@ function augmentGraphInfo(data) {
   return data;
 }
 
+function patchGraphData(m,graphInfo) {
+  m.nodeInfo = graphInfo.nodes;
+  const links = graphInfo.links;
+
+  // compute graph delta, for smooth transitions we need to mutate objects in-place
+  const nodeIdsToAdd = new Set(Object.keys(m.nodeInfo));
+  const nodeIdsToRemove = new Set();
+  m.data.nodes.forEach(node => {
+    if (nodeIdsToAdd.has(node.id)) {
+      nodeIdsToAdd.delete(node.id);
+    } else {
+      nodeIdsToRemove.add(node.id);
+    }
+  });
+  // apply the delta
+  nodeIdsToRemove.forEach(id => {
+    const index = m.data.nodes.findIndex(n => n.id == id);
+    m.data.nodes.splice(index, 1); // delete the element
+  });
+  nodeIdsToAdd.forEach(nodeId => {
+    m.data.nodes.push({
+      id: nodeId,
+    });
+  });
+  m.data.links = links; // links can be swapped out without problem
+
+  // check that selected/hovered nodes are still valid (see #397)
+  m.hoverNode = m.nodeInfo[m.hoverNode] != null ? m.hoverNode : null;
+  m.selectedNodes = new Set(
+    Array.from(m.selectedNodes).filter(nId => m.nodeInfo[nId] != null)
+  );
+
+  // annoying we need to call this function, but I haven't found a good workaround
+  graph.graphData(m.data);
+}
+
 function getNodeColor(nodeId, model) {
   const info = model.nodeInfo[nodeId];
   const style = model.style;
-  const typeFill = d3.rgb(
+  const typeFill = info.properties.color ? d3.rgb(info.properties.color) : d3.rgb(
```

Comment:
> does this mean that if a note has a `color` property it will show in that color within the graph?

Yes.

> also, for this operator we use the formatting:
> 
> ```
> const x = condition
>   ? when_true
>   : when_false
> ```

Am I not using that right? I thought that's what I was doing.